controller/support_bundle_controller.go

```diff
@@ -234,11 +233,19 @@ func (c *SupportBundleController) handleStatusUpdate(record *supportBundleRecord
 
 	isStatusChange := !reflect.DeepEqual(existing.Status, supportBundle.Status)
 	if isStatusChange {
-		supportBundle, err = c.ds.UpdateSupportBundleStatus(supportBundle)
-		if apierrors.IsConflict(errors.Cause(err)) {
+		_supportBundle, err := c.ds.UpdateSupportBundleStatus(supportBundle)
+		if err != nil && apierrors.IsConflict(errors.Cause(err)) {
 			log.WithError(err).Warnf(SupportBundleMsgRequeueOnConflictFmt, supportBundle.Name)
 			c.enqueue(supportBundle)
 		}
+		// Avoid requeue if the object is nil to avoid unnecessary requeue loops.
+		// This can happen if the update failed, for example, when the object was
+		// deleted during reconciliation.
+		if _supportBundle == nil {
+			log.WithError(err).Warnf("SupportBundle %v is nil after update, skipping requeue", supportBundle.Name)
+			return
+		}
```

Comment:
If `_supportBundle` is nil, why isn't a not found error returned?

Reply:
If the object is not found, an error will be returned. However, there could potentially be other kinds of errors that result in a nil object, so we will remove it from the queue.
https://github.com/longhorn/longhorn-manager/blob/548229725656086ef4e6387038ed340616ed7810/datastore/longhorn.go#L5482